jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -91,15 +91,40 @@ class SpectrumPerSpaxel(ProfileFromCube):
     action_text = 'See spectrum at a single spaxel'
     tool_tip = 'Click on the viewer and see the spectrum at that spaxel in the spectrum viewer'
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_moving = False
+
+    def activate(self):
+        super().activate()
+        for k in ("y_min", "y_max"):
+            self._profile_viewer.state.add_callback(k, self.on_limits_change)
+
+    def deactivate(self):
+        for k in ("y_min", "y_max"):
+            self._profile_viewer.state.remove_callback(k, self.on_limits_change)
+        super().deactivate()
+
+    def on_limits_change(self, *args):
+        if not self._is_moving:
+            self._previous_bounds = self._profile_viewer.get_limits()
+
     def on_mouse_move(self, data):
+        self._is_moving = False
         if data['event'] == 'mouseleave':
             self._mark.visible = False
             self._reset_profile_viewer_bounds()
             return
 
-        x = int(np.round(data['domain']['x']))
-        y = int(np.round(data['domain']['y']))
+        self._is_moving = True
+        try:
+            x = int(np.round(data['domain']['x']))
+            y = int(np.round(data['domain']['y']))
+            self._mouse_move_worker(x, y)
+        finally:
+            self._is_moving = False
```

Comment:
is it possible to send a limits change in before a mouseleave?  I guess maybe from API commands submitted in a still running cell or something (or would those just block events anyways)?

Reply:
Anything is possible via API, but in practice, will a user actually do that?

Reply:
if not, then we probably don't have to reset to `False` until the `mouseleave` event, right?

Reply:
Better be safe than sorry? It might be in a dirty state because I cannot guarantee otherwise.